File: src/main/java/com/practica/crean/arquitecture/infraestructure/delivery/dto/impl/EstudianteMapperDto.java
package com.practica.crean.arquitecture.infraestructure.delivery.dto.impl;

import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.infraestructure.delivery.dto.EstudianteResponse;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(componentModel = "spring")
public interface EstudianteMapperDto {

    @Mapping(expression = "java(estudiante.getNombre() + ' ' + estudiante.getApellidos())", target = "nombreCompleto")
    EstudianteResponse toResponse(Estudiante estudiante);

    List<EstudianteResponse> toListaEstudiantes(List<Estudiante> estudiantes);

}

File: src/main/kotlin/com/practica/crean/arquitecture/infraestructure/delivery/dto/mapper/EstudianteMapperDto.kt
package com.example.demo.infraestructure.delivery.dto.mapper

import com.example.demo.core.domain.Estudiante
import com.example.demo.infraestructure.delivery.dto.EstudianteResponse
import org.mapstruct.Mapper
import org.mapstruct.Mapping

// <>
@Mapper(componentModel = "spring")
interface EstudianteMapperDto {

    @Mapping(expression = "java(estudiante.getNombre() + ' ' + estudiante.getApellidos())", target = "nombreCompleto")
    fun toEstudianteResponse(estudiante: Estudiante): EstudianteResponse

    fun toListaEstudiantesResponse(estudiantes: List<Estudiante>): List<EstudianteResponse>

}